aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java

```diff
@@ -386,6 +390,22 @@ private void getSpiesSimulateConnection(final ChannelUri channelUri, final Media
         spiesSimulateConnection = null != sscStr ? "true".equals(sscStr) : ctx.spiesSimulateConnection();
     }
 
+    private void getUntetheredWindowLimitTimeout(final ChannelUri channelUri, final MediaDriver.Context ctx)
+    {
+        final String timeoutString = channelUri.get(UNTETHERED_WINDOW_LIMIT_TIMEOUT_PARAM_NAME);
+        untetheredWindowLimitTimeoutNs = null != timeoutString ?
+            SystemUtil.parseDuration(UNTETHERED_WINDOW_LIMIT_TIMEOUT_PARAM_NAME, timeoutString) :
+            ctx.untetheredWindowLimitTimeoutNs();
+    }
+
+    private void getUntetheredRestingTimeout(final ChannelUri channelUri, final MediaDriver.Context ctx)
+    {
+        final String timeoutString = channelUri.get(UNTETHERED_RESTING_TIMEOUT_PARAM_NAME);
+        untetheredWindowLimitTimeoutNs = null != timeoutString ?
+            SystemUtil.parseDuration(UNTETHERED_RESTING_TIMEOUT_PARAM_NAME, timeoutString) :
+            ctx.untetheredWindowLimitTimeoutNs();
+    }
+
```

Comment:
Could be refactored to a single method similar to `PublicationImage.getTimeoutNsFromChannel`. Maybe we could add a method to `ChannelUri`?

Reply:
I've done the refactoring, but I'm going to leave it in the current place for now.